keeper/component/dispatcher/genesis_test.go

```diff
@@ -0,0 +1,219 @@
+					*defaultAmounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-13", "chain-14"),
+				)
+				g.DispatchedCounts = append(
+					g.DispatchedCounts,
+					*defaultCounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-1", "chain-2"),
+					*defaultCounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-3", "chain-4"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-11", "chain-12"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-13", "chain-14"),
+					*defaultCounts(core.PROTOCOL_HYPERLANE, core.PROTOCOL_CCTP, "1", "2"),
+					*defaultCounts(core.PROTOCOL_HYPERLANE, core.PROTOCOL_CCTP, "3", "4"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_HYPERLANE, "11", "12"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_HYPERLANE, "13", "14"),
+				)
+
+				return g
+			},
+			expErr: "",
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			d, deps := mocks.NewDispatcherComponent(t)
+			ctx := deps.SdkCtx
+
+			g := tC.genesis()
+			err := d.InitGenesis(ctx, g)
+
+			if tC.expErr != "" {
+				require.ErrorContains(t, err, tC.expErr)
+			} else {
+				require.NoError(t, err)
+				require.ElementsMatch(t, g.DispatchedAmounts, d.GetAllDispatchedAmounts(ctx))
+				require.ElementsMatch(t, g.DispatchedCounts, d.GetAllDispatchedCounts(ctx))
+			}
+		})
+	}
+}
+
+func TestExportGenesis(t *testing.T) {
+	testCases := []struct {
+		name              string
+		dispatchedAmounts func() []dispatchertypes.DispatchedAmountEntry
+		dispatchedCounts  func() []dispatchertypes.DispatchCountEntry
+		expEmpty          bool
+	}{
+		{
+			name:     "success - empty state",
+			expEmpty: true,
+		},
+		{
+			name: "success - not empty state",
+			dispatchedAmounts: func() []dispatchertypes.DispatchedAmountEntry {
+				return []dispatchertypes.DispatchedAmountEntry{
+					*defaultAmounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-1", "chain-2"),
+					*defaultAmounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-3", "chain-4"),
+					*defaultAmounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-11", "chain-12"),
+					*defaultAmounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-13", "chain-14"),
+				}
+			},
+			dispatchedCounts: func() []dispatchertypes.DispatchCountEntry {
+				return []dispatchertypes.DispatchCountEntry{
+					*defaultCounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-1", "chain-2"),
+					*defaultCounts(core.PROTOCOL_IBC, core.PROTOCOL_CCTP, "chain-3", "chain-4"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-11", "chain-12"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_IBC, "chain-13", "chain-14"),
+					*defaultCounts(core.PROTOCOL_HYPERLANE, core.PROTOCOL_CCTP, "1", "2"),
+					*defaultCounts(core.PROTOCOL_HYPERLANE, core.PROTOCOL_CCTP, "3", "4"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_HYPERLANE, "11", "12"),
+					*defaultCounts(core.PROTOCOL_CCTP, core.PROTOCOL_HYPERLANE, "13", "14"),
+				}
+			},
```

Comment:
why are these defined as a function if they're only returning `[]dispatchertypes.DispatchCountEntry`? could just use the slice type in the field